osquery/tables/events/linux/bpf_file_events.cpp

```diff
@@ -0,0 +1,179 @@
+    auto parser = Config::getParser("file_paths");
+    if (parser == nullptr) {
+        LOG(ERROR) << "No file paths found";
+        return;
+    }
+    auto const& doc = parser->getData().doc();
+    auto file_accesses_it = doc.FindMember("file_accesses");
+    if (file_accesses_it == doc.MemberEnd()) {
+    LOG(ERROR) << "No key 'file_accesses' found when parsing file events"
+                    " subscriber configuration.";
+    return;
+    }
+    auto& accesses = file_accesses_it->value;
+    if (accesses.GetType() != rapidjson::kArrayType) {
+    LOG(ERROR) << "Wrong type found for file_accesses when parsing file events"
+                    " subscriber configuration. Found ("
+                << accesses.GetType()
+                << "),"
+                    " expected array ("
+                << rapidjson::kArrayType << ").";
+    return;
+    }
+
+    Config::get().files([&accesses](const std::string& category,
+                const std::vector<std::string>& files) {
+                for (auto file : files) {
+                    replaceGlobWildcards(file);
+                    //std::cout << "File pattern: " << file << std::endl;
+
+                    std::vector<std::string> file_path_list = {};
+                    resolveFilePattern(file, file_path_list);
+
+                    //for (const auto& file_path : file_path_list) {
+                    //    std::cout << "\t File path: " << file_path << std::endl;
+                    //}
+
+                    monitored_paths.insert(monitored_paths.end(), file_path_list.begin(), file_path_list.end());
+                }
+    });
+}
+
+void BPFFileEventSubscriber::configure() {
+    configure_filtering();
+}
+
+Status BPFFileEventSubscriber::init() {
+  auto subscription_context = createSubscriptionContext();
+  subscribe(&BPFFileEventSubscriber::eventCallback, subscription_context);
+
+  return Status::success();
+}
+
+Status BPFFileEventSubscriber::eventCallback(const ECRef& event_context,
+                                               const SCRef&) {
+  auto row_list = generateRowList(event_context->event_list);
+  addBatch(row_list);
+
+  return Status::success();
+}
+
+bool BPFFileEventSubscriber::generateRow(
+    Row& row, const ISystemStateTracker::Event& event) {
+  row = {};
+
+  switch (event.type) {
+  case ISystemStateTracker::Event::Type::Open:
+    row["syscall"] = TEXT("open");
+    break;
+
+  default:
+    return false;
+  }
+
+  bool is_path_matched = false;
+
+  if (!std::holds_alternative<ISystemStateTracker::Event::FileIOData>(event.data)) {
+    row["file_path"] = TEXT("");
+    row["flags"] = TEXT("");
+
+    } else {
+      const auto& file_io_data =
+        std::get<ISystemStateTracker::Event::FileIOData>(event.data);
+
+      row["file_path"] = TEXT(file_io_data.file_path);
+      row["flags"] = TEXT(file_io_data.flags);
+
+      if (file_io_data.flags.find("O_CREAT") != std::string::npos)
+      {
+          configure_filtering();
```

Comment:
Configuration should only be read once, within the dedicated ::configure() method.

Additionally, triggering a complete path expansion whenever a new file is created is not a good strategy (refer to the comment left on `configure_filtering`)